### ISSUE TITLE
extend keyword in exported client schema

### ISSUE THREAD

**Issue by atoy40** [dotansimha/graphql-code-generator#2134]
**Describe the bug**
The tool does not seems to support the "extend" keyword used to extend type (for example in client schema). It seems schemas are loaded, but the extended Query/Mutation are not merged with the server schema which contains main Query/Mutation types. So the tools crash when generating graphql operations because It knows nothing about "extended" query/mutation fields.

If I remove "extend" it wworks, but then I've other problems with the apollo client.

**To Reproduce**

Add the extend keyword on the Query or Mutation type of local schema.

1. My GraphQL schema:

Server schema :

```graphql
Query {
  serverbool: Boolean
  serverstring: String
}
```

Client only schema :

```graphql
extend Query {
  localbool: Boolean
  localstring: String
}
```

2. My GraphQL operations:

```graphql
query {
   localbool @client
}
```
The generation of operations failed because it cannot find field "localbool" on type "Query"

3. My `codegen.yml` config file:

```json
{
"./web/src/graphql-type-hooks.ts": {
      "schema": ["./server/src/schema.ts", "./web/src/client.ts"],
      "documents": "./web/src/graphql/**/*.ts",
      "config": {
        "withHOC": false,
        "withComponent": false,
        "withHooks": true,
        "withMutationFn": false
      },
      "plugins": ["typescript", "typescript-operations", "typescript-react-apollo"]
    }
}
```

**Environment:**

- OS: 
- `@graphql-codegen/...`: 1.3.1
- NodeJS: 

**Additional context**

**Comment by ardatan** [comment]
Could you create a reproduction repo?

**Comment by atoy40** [comment]
sandbox : https://codesandbox.io/s/graphql-codegen-issue-template-1vip9

Hello, I started from the template and just added clientschema.graphql to simulate apollo state "client only" schema. It extends Query with a boolean field (modalOpen). I've also added an operation that query this field with the @client tag. When you execute yarn codegen, it crash with this error :

```GraphQLDocumentError: Cannot query field "modalOpen" on type "Query". Did you mean "modelOpen"?```

However, the types.ts file looks good because the Query type describes both schema et localschema Query fields :
```typescript
export type Query = {
  __typename?: "Query";
  user: User;
  modelOpen: Scalars["Boolean"];
};
```

**Comment by ardatan** [comment]
First, the error looks correct  because you request for modalOpen in document while it is defined as modelOpen in your schema.
Second, I couldn’t see anything wrong with ‘extend’ in here. Codegen extends Query type as you defined.
Am I missing something?

**Comment by atoy40** [comment]
Oops, you're right about my mistake, I fixed it, and yes it works ! so I mimic my real case a bit more in the sandbox. The problem appears when the client schema (the one with extend) is coming from the export of a typescript (and probably js too) file. But may be it's not a problem with codegen, but with graphql-tag or graphql-toolkit ?

https://codesandbox.io/s/graphql-codegen-issue-template-1vip9

**Comment by ardatan** [comment]
@atoy40 `types.ts` looks correct, because it has `modalOpen` from client schema.

**Comment by atoy40** [comment]
@ardatan did you run yarn codegen ? the types.ts is one generated at a previous run without the extend keyword.

**Comment by ardatan** [comment]
@atoy40 Now I see.
So could you try again with the following versions?
```
 - @graphql-codegen/cli => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/core => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/flow => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/flow-operations => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/flow-resolvers => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/java-common => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/java => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/java-resolvers => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/add => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/fragment-matcher => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/introspection => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/schema-ast => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/time => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/visitor-plugin-common => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-apollo-angular => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-compatibility => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-document-nodes => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-graphql-files-modules => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-mongodb => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-operations => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-react-apollo => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-resolvers => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-stencil-apollo => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/typescript-urql => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/near-operation-file-preset => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/testing => 1.3.1-alpha-13087229.95+13087229
 - @graphql-codegen/plugin-helpers => 1.3.1-alpha-13087229.95+13087229
```

**Comment by atoy40** [comment]
@ardatan looks perfect to me :) I've tested it in my project, no more errors and typescript types have both server and local "extended" fields.

**Comment by dotansimha** [comment]
Available in 1.4.0 🎉
